packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -2662,22 +2675,23 @@ function updateDehydratedSuspenseComponent(
     // This boundary is in a permanent fallback state. In this case, we'll never
     // get an update and we'll never be able to hydrate the final content. Let's just try the
     // client side render instead.
-    const {errorMessage} = getSuspenseInstanceFallbackErrorDetails(
+    const {message, hash, stack} = getSuspenseInstanceFallbackErrorDetails(
```

Comment:
Does this intermediate object get DCE?

Reply:
I think you got the tradeoff right. It's just unfortunate that Closure isn't smart enough. Seems like it should be able to.

Reply:
Maybe if you always returned the same object with the same signature at the end and just assigned to local variables before it?

Reply:
Yeah I got this to work